File: source/includes/write/delete.kt
import com.mongodb.client.model.DeleteOptions
import com.mongodb.client.model.Filters.*
import com.mongodb.kotlin.client.MongoClient

// start-data-class
data class Restaurant(val name: String, val borough: String)
// end-data-class

fun main() {
    val uri = "<connection string>"

    val mongoClient = MongoClient.create(uri)
    val database = mongoClient.getDatabase("sample_restaurants")
    val collection = database.getCollection<Movie>("restaurants")

    // start-delete-one
    val filter = eq(Restaurant::name.name, "Happy Garden")
    val result = collection.deleteOne(filter)
    // end-delete-one

    // start-delete-many
    val filter = and(
        eq(Restaurant::borough.name, "Brooklyn"),
        eq(Restaurant::name.name, "Starbucks")
    )
    val result = collection.deleteMany(filter)
    // end-delete-many

    // start-delete-options
    val opts = DeleteOptions().comment("sample comment")
    val filter = regex(Restaurant::name.name, "Red")
    val result = collection.deleteOne(filter, opts)
    // end-delete-options

}

File: source/includes/write/delete.scala
import org.mongodb.scala._
import org.mongodb.scala.bson.Document
import org.mongodb.scala.model.Filters.{and, equal, regex}
import org.mongodb.scala.model.{DeleteOptions}
import org.mongodb.scala.result.{DeleteResult}

object Delete {

  def main(args: Array[String]): Unit = {
    val mongoClient = MongoClient("<connection string URI>")

    // start-db-coll
    val database: MongoDatabase = mongoClient.getDatabase("sample_restaurants")
    val collection: MongoCollection[Document] = database.getCollection("restaurants")
    // end-db-coll

    // start-delete-one
    val filter = equal("name", "Happy Garden")
    val observable: Observable[DeleteResult] = collection.deleteOne(filter)

    observable.subscribe(new Observer[DeleteResult] {
      override def onNext(result: DeleteResult): Unit = println(s"Deleted document count: ${result.getDeletedCount}")
      override def onError(e: Throwable): Unit = println(s"Error: $e")
      override def onComplete(): Unit = println("Completed")
//    })
    // end-delete-one

    // start-delete-many
    val filter = and(
      equal("borough", "Brooklyn"),
      equal("name", "Starbucks")
    )
    val observable: Observable[DeleteResult] = collection.deleteMany(filter)

    observable.subscribe(new Observer[DeleteResult] {
      override def onNext(result: DeleteResult): Unit = println(s"Deleted document count: ${result.getDeletedCount}")
      override def onError(e: Throwable): Unit = println(s"Error: $e")
      override def onComplete(): Unit = println("Completed")
    })
    // end-delete-many

    // start-delete-options
    val filter = regex("name", "Red")
    val opts = DeleteOptions().comment("sample comment")
    val observable = collection.deleteMany(filter, opts)

    observable.subscribe(new Observer[DeleteResult] {
      override def onNext(result: DeleteResult): Unit = println(s"Deleted document count: ${result.getDeletedCount}")
      override def onError(e: Throwable): Unit = println(s"Error: $e")
      override def onComplete(): Unit = println("Completed")
    })
    // end-delete-options

    Thread.sleep(1000)
    mongoClient.close()
  }
}
